### DISCUSSION TITLE
Is it possible to create worktree with detached head?

### DISCUSSION THREAD

**Discussion by akavel** [libgit2/libgit2#6720] [Q&A]
I want to create a worktree for an existing directory - but I don't want to add a branch in it. As far as I understand, in git CLI I can do this with `git worktree add --detach PATH`. Is there a way to do that in libgit2, or am I misunderstanding how that CLI feature works?

I understand to create a worktree in libgit2 I need to use [git_worktree_add()](https://libgit2.org/libgit2/#HEAD/group/worktree/git_worktree_add). However, by default it seems to create a new branch based on the passed `name` argument. Is it possible to call it in some way that would not create a new branch, and just use HEAD or any passed reference in a detached mode?

**Comment by LucasKuhlemann** [comment]
Did you find a way to accomplish this?
